File: app/src/main/java/com/tinilite/cardxml/SignFragment.java
package com.tinilite.cardxml;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


public class SignFragment extends Fragment {


    public SignFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.fragment_sign, container, false);

        if (view.findViewById(R.id.card1) != null) {

            // However, if we're being restored from a previous state,
            // then we don't need to do anything and should return or els
            //
            // we could end up with overlapping fragments.
            if (savedInstanceState != null) {
                return view;
            }
            // Add the fragment to the 'fragment_container' FrameLayout
            getFragmentManager().beginTransaction()
                    .add(R.id.card1, CardFragment.newInstance(1)).commit();

            getFragmentManager().beginTransaction()
                    .add(R.id.card2, CardFragment.newInstance(2)).commit();

        }

        return view;
    }
}


/**
 * A simple {@link Fragment} subclass.
 * Use the {@link SignFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
/*
/ TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private static final String ARG_PARAM1 = "param1";
private static final String ARG_PARAM2 = "param2";

// TODO: Rename and change types of parameters
private String mParam1;
private String mParam2;

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment SignFragment.

// TODO: Rename and change types and number of parameters
public static SignFragment newInstance(String param1, String param2) {
    SignFragment fragment = new SignFragment();
    Bundle args = new Bundle();
    args.putString(ARG_PARAM1, param1);
    args.putString(ARG_PARAM2, param2);
    fragment.setArguments(args);
    return fragment;
}

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

        *
 */

File: app/src/main/java/com/tinilite/cardxml/SignFragment.kt
package com.tinilite.cardxml

import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

class SignFragment() : Fragment(), Parcelable {
    private var mParam1: String? = null
    private var mParam2: String? = null

    constructor(parcel: Parcel) : this() {
        mParam1 = parcel.readString()
        mParam2 = parcel.readString()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mParam1 = arguments!!.getString(ARG_PARAM1)
            mParam2 = arguments!!.getString(ARG_PARAM2)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_sign, container, false)
    }

    companion object {
        private const val ARG_PARAM1 = "param1"
        private const val ARG_PARAM2 = "param2"
        fun newInstance(param1: String?, param2: String?): SignFragment {
            val fragment = SignFragment()
            val args = Bundle()
            args.putString(ARG_PARAM1, param1)
            args.putString(ARG_PARAM2, param2)
            fragment.arguments = args
            return fragment
        }
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(mParam1)
        parcel.writeString(mParam2)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SignFragment> {
        override fun createFromParcel(parcel: Parcel): SignFragment {
            return SignFragment(parcel)
        }

        override fun newArray(size: Int): Array<SignFragment?> {
            return arrayOfNulls(size)
        }
    }
}